src/main/java/bio/terra/app/controller/DataRepositoryServiceApiController.java

```diff
@@ -104,10 +105,17 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     @Override
     public ResponseEntity<DRSObject> getObject(
         @PathVariable("object_id") String objectId,
-        @RequestParam(value = "expand", required = false, defaultValue = "false") Boolean expand) {
-        // The incoming object id is a DRS object id, not a file id.
-        AuthenticatedUserRequest authUser = getAuthenticatedInfo();
-        return new ResponseEntity<>(drsService.lookupObjectByDrsId(authUser, objectId, expand), HttpStatus.OK);
+        @RequestParam(value = "expand", required = false, defaultValue = "false") Boolean expand
+    ) {
+        try {
+            // The incoming object id is a DRS object id, not a file id.
+            AuthenticatedUserRequest authUser = getAuthenticatedInfo();
+            drsService.increment();
```

Comment:
Where should this live then? I'm at a loss about a better way to set this up

Reply:
Looking at the change history, you originally had the `increment` and `decrement` calls inside `DrsService. lookupObjectByDrsId()`. This seems like a better place to me. Is there a reason why you changed this?

Putting the check inside the service call is better IMO because it preserves the abstraction boundary around the behavior of `lookupObjectByDrsId`. It's also DRY-er in that, if you have many calls to this method, you'd need to add the inc/dec calls around each one. Putting it in the class code avoids this code duplication.